### PR TITLE
feat: add poddisruptionbudget

### PR DESCRIPTION
**Pull request by marbonilla**
# Description:
This PR add PodDisruptionBudget for `caddy|lms|lms-worker|cms|cms-worker|forum|mfe` deployments

## How to test it?
You can install the last version of drydock and include variable `DRYDOCK_ENABLE_PDB = true` in `config.yml` file, run `tutor config save` or `make config` and syncronize.
For verify PDBs, you can run in your namespace `kubectl get pdb` and you can see:
``` bash
NAME             MIN AVAILABLE   MAX UNAVAILABLE   ALLOWED DISRUPTIONS   AGE
caddy-pdb        30%             N/A               2                     11m
cms-pdb          30%             N/A               2                     11m
cms-worker-pdb   30%             N/A               2                     11m
forum-pdb        30%             N/A               2                     11m
lms-pdb          30%             N/A               2                     11m
lms-worker-pdb   30%             N/A               2                     11m
mfe-pdb          30%             N/A               2                     11m
```
You can verify its operation in the Yamato cluster in the [quince namespace](https://argocd.services.stage.edunext-shipyard.net/applications/argocd/quince.yamato?view=tree&conditions=false&resource=)

### CODE REVIEW

**Review by Ian2012 — APPROVED** [review]
LGTM

File: drydock/templates/drydock/k8s/patches/pdb.yml [hunk 1]
```diff
@@ -0,0 +1,74 @@
+{%- if MFE_HOST is defined %}
+---
+apiVersion: policy/v1
+kind: PodDisruptionBudget
+metadata:
+  name: mfe-pdb
+spec:
+  minAvailable: 30%
```
Comment (MoisesGSalas) [inline drydock/templates/drydock/k8s/patches/pdb.yml:8]: This value should be configurable.

Also I think we should be able to turn off the PDBs individually. We could probably take a similar approach to open-craft's Grove where certain values of `minAvailable` disable the pdb (i.e. `-1`).

**Review by Ian2012 — APPROVED** [review]

File: drydock/patches/kustomization-resources [hunk 2]
```diff
@@ -1,4 +1,5 @@
 - plugins/drydock/k8s/multipurpose-jobs.yml
+- plugins/drydock/k8s/patches/pdb.yml
```
Comment (jfavellar90) [inline drydock/patches/kustomization-resources:2]: I would not include this file in the `patches` folder. This file is creating new PDB resources so I would use a path like `plugins/drydock/k8s/pdb/openedx-pdb.yml`

File: drydock/templates/drydock/k8s/patches/pdb.yml [hunk 3]
```diff
@@ -0,0 +1,84 @@
+{%- if MFE_HOST is defined and DRYDOCK_PDB_MINAVAILABLE_PERCENTAGE_MFE != 0 %}
```
Comment (jfavellar90) [inline drydock/templates/drydock/k8s/patches/pdb.yml:1]:
```suggestion
{%- if MFE_HOST is defined and DRYDOCK_PDB_MINAVAILABLE_PERCENTAGE_MFE > 0 %}
```
I think it's better to use `greater than`. Please replace where it applies

**Review by jfavellar90 — APPROVED** [review]

**Review by Henrrypg — APPROVED** [review]

**Pull request merged**